consensus/consensus-types/src/vote_proposal.rs

```diff
@@ -13,66 +13,49 @@ use std::{
 
 /// This structure contains all the information needed by safety rules to
 /// evaluate a proposal / block for correctness / safety and to produce a Vote.
-#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive(Deserialize, Serialize)]
 pub struct VoteProposal<T> {
+    /// Contains the data necessary to construct the parent's execution output state
+    /// and the childs in a verifiable way
+    accumulator_extension_proof: AccumulatorExtensionProof<TransactionAccumulatorHasher>,
     /// The block / proposal to evaluate
     #[serde(bound(deserialize = "Block<T>: Deserialize<'de>"))]
     block: Block<T>,
-    /// The accumulator root hash after executing this block.
-    executed_state_id: HashValue,
-    /// The version of the latest transaction in the ledger.
-    version: Version,
     /// An optional field containing the set of validators for the start of the next epoch
     next_validator_set: Option<ValidatorSet>,
 }
 
 impl<T> VoteProposal<T> {
     pub fn new(
+        accumulator_extension_proof: AccumulatorExtensionProof<TransactionAccumulatorHasher>,
         block: Block<T>,
-        executed_state_id: HashValue,
-        version: Version,
         next_validator_set: Option<ValidatorSet>,
     ) -> Self {
         Self {
+            accumulator_extension_proof,
             block,
-            executed_state_id,
-            version,
             next_validator_set,
         }
     }
 
-    pub fn block(&self) -> &Block<T> {
-        &self.block
+    pub fn accumulator_extension_proof(
+        &self,
+    ) -> &AccumulatorExtensionProof<TransactionAccumulatorHasher> {
+        &self.accumulator_extension_proof
     }
 
-    pub fn executed_state_id(&self) -> HashValue {
-        self.executed_state_id
+    pub fn block(&self) -> &Block<T> {
+        &self.block
     }
 
     pub fn next_validator_set(&self) -> Option<&ValidatorSet> {
         self.next_validator_set.as_ref()
     }
-
-    pub fn version(&self) -> Version {
-        self.version
-    }
 }
 
 impl<T: PartialEq> Display for VoteProposal<T> {
     fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
-        write!(
-            f,
-            "VoteProposal[block: {}, executed_state_id: {}, version: {}, next_validator_set: {}]",
-            self.block,
-            self.executed_state_id,
-            self.version,
-            self.next_validator_set
```

Comment:
This is not needed any more?

Reply:
It doesn't seem all that relevant since the block has the information we need. Also the next_validator_set is expected to be long and not particularly interesting for non-debug purposes. We also lose the executed_state_id and version as we can compute those from the proof.